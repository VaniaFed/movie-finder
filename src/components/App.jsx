import React, { Component } from 'react';
import { v4 } from 'uuid';

import { WelcomeTitle } from './ui/WelcomeTitle';
import { Menu } from './ui/Menu';
import { FormCreateMenuItem } from './ui/FormCreateMenuItem';
import { MyLink } from './ui/MyLink';

export class App extends Component {
    state = {
        menuList: [
            {
                id: v4(),
                title: 'First'
            },
            {
                id: v4(),
                title: 'Second'
            },
            {
                id: v4(),
                title: 'Third'
            }
        ],
        inputValue: ''
    };

    createMenuItem = () => {
        const { inputValue } = this.state;
        this.setState(({ menuList }) => ({
            menuList: [
                ...menuList,
                {
                    id: v4(),
                    title: inputValue
                }
            ]
        }));
    };
    changeInputValue = newInputValue => {
        this.setState(() => ({ inputValue: newInputValue }));
    };

    render() {
        const { menuList } = this.state;
        const { createMenuItem, changeInputValue } = this;
        return (
            <div className="App">
                <WelcomeTitle title="Hello, world" />
                <Menu menuList={menuList} />
                <FormCreateMenuItem
                    onClick={createMenuItem}
                    changeInputValue={changeInputValue}
                />
                <MyLink text="I'm in VK" reference="https://vk.com/vaniafed" />
            </div>
        );
    }
}
