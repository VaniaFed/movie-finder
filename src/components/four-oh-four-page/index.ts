export { FourOfFourPage } from './four-oh-four-page';
