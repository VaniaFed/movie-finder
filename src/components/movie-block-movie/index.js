export { MovieBlockMovie } from './movie-block-movie';
