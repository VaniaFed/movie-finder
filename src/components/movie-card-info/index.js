export { MovieCardInfo } from './movie-card-info';
