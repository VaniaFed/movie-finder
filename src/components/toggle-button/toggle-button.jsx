import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './toggle-button.scss';

export const ToggleButton = ({
    label,
    className,
    checked,
    name,
    background
}) => {
    console.log(background);
    const resultClass = classNames(className, 'toggle-button', {
        'toggle-button--no--background': !background
    });
    return (
        <label className={resultClass}>
            <input
                className={classNames('toggle-button__input', {
                    'toggle-button__input--no--background': !background
                })}
                type="radio"
                defaultChecked={checked || ''}
                name={name}
            />
            <span
                className={classNames('toggle-button__text', {
                    'toggle-button__text--no--background': !background
                })}
            >
                {label}
            </span>
        </label>
    );
};
ToggleButton.propTypes = {
    label: PropTypes.string.isRequired,
    className: PropTypes.string,
    checked: PropTypes.string,
    name: PropTypes.string.isRequired
};
