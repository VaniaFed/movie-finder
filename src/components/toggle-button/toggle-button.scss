@import '~_vars.scss';
.toggle-button {
    margin-left: 10px;
    border-radius: 4px;
    overflow: auto;
    float: left;
    cursor: pointer;
    &__text {
        text-align: center;
        display: block;
        cursor: pointer;
        text-transform: uppercase;
        padding: 8px 20px;
        background: #444;
        color: #fff;
        &#{&}--no--background {
            background: none;
            text-transform: initial;
            color: #000;
            padding: 0;
        }
    }
    &__input {
        position: absolute;
        top: -500px;
        &:checked + .toggle-button__text {
            background: $accent-color;
        }
        &#{&}--no--background:checked + .toggle-button__text {
            background: none;
            color: $accent-color;
        }
    }
}
