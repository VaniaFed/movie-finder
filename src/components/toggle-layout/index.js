export { ToggleLayout } from './toggle-layout';
