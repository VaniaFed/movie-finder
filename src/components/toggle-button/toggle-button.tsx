import React from 'react';
import classNames from 'classnames';
import { Props } from './props';

import './toggle-button.scss';

export const ToggleButton: React.FC<Props> = ({
    label,
    className,
    name,
    background,
    checked,
    onClick
}: Props) => {
    const resultClass: string = classNames(className, 'toggle-button', {
        'toggle-button--no--background': !background
    });
    const inputClass: string = classNames('toggle-button__input', {
        'toggle-button__input--no-background': !background,
        'toggle-button__input--checked': checked
    });
    const spanClass: string = classNames('toggle-button__text', {
        'toggle-button__text--no--background': !background,
        'toggle-button__text--checked': checked,
        'toggle-button__text--no--background--checked': checked && !background
    });

    return (
        <label className={resultClass} onClick={onClick}>
            <input
                className={inputClass}
                type="radio"
                defaultChecked={checked && 'checked'}
                name={name}
            />
            <span className={spanClass}>{label}</span>
        </label>
    );
};
