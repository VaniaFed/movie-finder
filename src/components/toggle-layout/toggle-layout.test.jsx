import renderer from 'react-test-renderer';
import { ToggleButton } from 'components/toggle-button';
import { ToggleLayout } from './index';

const data = [
    { key: 'hello_world', text: 'Hello world!' },
    { key: 'we_are_strong', text: 'We are stong' }
];
const buttons = data.map(item => (
    <ToggleButton
        key={item.key}
        checked={item.key}
        label={item.text}
        name="test"
        onClick={f => f}
        background
    />
));
describe('<ToggleLayout />', () => {
    const tree = renderer
        .create(<ToggleLayout>{buttons}</ToggleLayout>)
        .toJSON();
    it('should be rendered', () => {
        expect(tree).toMatchSnapshot();
    });
});
