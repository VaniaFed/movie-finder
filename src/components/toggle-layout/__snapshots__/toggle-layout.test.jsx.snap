// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleLayout /> should be rendered 1`] = `
<div>
  <label
    className="toggle-button"
    onClick={[Function]}
  >
    <input
      className="toggle-button__input toggle-button__input--checked"
      defaultChecked="checked"
      name="test"
      type="radio"
    />
    <span
      className="toggle-button__text toggle-button__text--checked"
    >
      Hello world!
    </span>
  </label>
  <label
    className="toggle-button"
    onClick={[Function]}
  >
    <input
      className="toggle-button__input toggle-button__input--checked"
      defaultChecked="checked"
      name="test"
      type="radio"
    />
    <span
      className="toggle-button__text toggle-button__text--checked"
    >
      We are stong
    </span>
  </label>
</div>
`;
